core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/action/ImportDictionaryServlet.java

```diff
@@ -0,0 +1,132 @@
+package be.orbinson.aem.dictionarytranslator.servlets.action;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.apache.tika.io.IOUtils;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.Servlet;
+
+@Component(service = Servlet.class)
+@SlingServletResourceTypes(
+        resourceSuperType = "granite/ui/components/coral/foundation/form",
+        resourceTypes = "aem-dictionary-translator/servlet/action/import-dictionary",
+        methods = "POST")
+public class ImportDictionaryServlet extends SlingAllMethodsServlet {
+    private static final Logger LOG = LoggerFactory.getLogger(ImportDictionaryServlet.class);
+
+    @Override
+    public void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
+        String path = request.getParameter("dictionary");
+        RequestParameter csvfile = request.getRequestParameter("./csvfile");
+
+        if (csvfile != null) {
+            InputStream csvContent = csvfile.getInputStream();
+
+            List<String> languages = new ArrayList<>();
+            List<String> KEYs = new ArrayList<>();
+            List<List<String>> translations = new ArrayList<>();
+
+            ResourceResolver resourceResolver = request.getResourceResolver();
+
+            try (InputStreamReader reader = new InputStreamReader(csvContent)) {
+                String result = IOUtils.toString(csvContent, String.valueOf(StandardCharsets.UTF_8));
+                csvContent.reset();
```

Comment:
why do you need this `reset()`?

Reply:
The InputStreamReader reader  is still needed to get the headers via the csvParser, this way the reader will reset to the beginning of the inputstream and the csvparser will be able to extract the headers